### PR TITLE
stm32/sdio.c: Add functions to disable/enable SDIOITIE and reenable SDIO.

### PR DESCRIPTION
**Pull request by iabdalkader**
* As per discussion in #6986.
* Tested on PYBD_SF2
* Note had to fix cyw43_ctrl.c formatting.

### CODE REVIEW

File: ports/stm32/sdio.c [hunk 1]
```diff
@@ -93,6 +93,21 @@ void sdio_deinit(void) {
     #endif
 }
 
+void sdio_reenable(void) {
+    if (__HAL_RCC_SDMMC1_IS_CLK_DISABLED()) {
+        __HAL_RCC_SDMMC1_CLK_ENABLE(); // enable SDIO peripheral
+        sdio_enable_high_speed_4bit();
```
Comment (iabdalkader) [inline ports/stm32/sdio.c:99]: should this be here or called after `sdio_reenable()` in cyw43_ctrl.c ?

Comment (dpgeorge) [reply ports/stm32/sdio.c:99]: I think keep it here (no functional change then).

**Comment by dpgeorge** [comment]
> Note had to fix cyw43_ctrl.c formatting.

You shouldn't need to do that, this file is not part of `codeformat.py` (it could be added, but that would be best for a separate PR...).

**Comment by iabdalkader** [comment]
> You shouldn't need to do that, this file is not part of `codeformat.py` (it could be added, but that would be best for a separate PR...).

It's the last commit I can reset it, but I thought we needed to run this on every changed file before sending the PR otherwise the tests fail ?

**Comment by dpgeorge** [comment]
> but I thought we needed to run this on every changed file before sending the PR otherwise the tests fail ?

The code in `drivers/` is not run through uncrustify at the moment, so formatting errors are allowed there :)

**Comment by iabdalkader** [comment]
> The code in `drivers/` is not run through uncrustify at the moment, so formatting errors are allowed there :)

I see, okay I reverted the codeformat, and will get around to updating the other PRs tomorrow.

**Comment by dpgeorge** [comment]
Merged in 80788154b31c6097fd152e6e2e971319e8e3da1e and d3eb6d68a377c42bd37e609ce667bea815fecd4f

**Pull request closed without merging**